### PR TITLE
[Drop-In UI] Update state when click on end navigation

### PR DESCRIPTION
**Pull request by abhishek1508**
### Description

Fixes https://github.com/mapbox/navigation-sdks/issues/1517

### CODE REVIEW

File: libnavui-dropin/src/main/java/com/mapbox/navigation/dropin/component/routeline/RouteLineComponent.kt [hunk 1]
```diff
@@ -42,11 +42,19 @@ internal class RouteLineComponent(
     }
 
     private val routesObserver = RoutesObserver { result ->
-        val routeLines = result.routes.map { RouteLine(it, null) }
+        val routeLines = result.navigationRoutes
         jobControl.scope.launch {
-            routeLineApi.setRoutes(routeLines).let { routeDrawData ->
+            if (routeLines.isNotEmpty()) {
```
Comment (cafesilencio) [inline libnavui-dropin/src/main/java/com/mapbox/navigation/dropin/component/routeline/RouteLineComponent.kt:47]: This is incorrect. The empty collection should be passed into `routeLineApi.setNavigationRoutes()`.

Comment (abhishek1508) [reply libnavui-dropin/src/main/java/com/mapbox/navigation/dropin/component/routeline/RouteLineComponent.kt]: Removed.

File: libnavui-dropin/src/main/java/com/mapbox/navigation/dropin/component/routefetch/RoutesViewModel.kt [hunk 2]
```diff
@@ -61,16 +61,11 @@ internal class RoutesViewModel(
                 } else {
                     startNavigation(mapboxNavigation)
                 }
+                return state.copy(navigationStarted = true)
```
[the remark this answers is not in the document]
Comment (abhishek1508) [reply libnavui-dropin/src/main/java/com/mapbox/navigation/dropin/component/routefetch/RoutesViewModel.kt]: Fair point. Fixed.

File: libnavui-dropin/src/main/java/com/mapbox/navigation/dropin/component/routefetch/RoutesViewModel.kt [hunk 3]
```diff
@@ -110,11 +108,12 @@ internal class RoutesViewModel(
 
     private fun stopNavigation(mapboxNavigation: MapboxNavigation) {
         with(mapboxNavigation) {
+            setNavigationRoutes(listOf())
```
Comment (cafesilencio) [inline libnavui-dropin/src/main/java/com/mapbox/navigation/dropin/component/routefetch/RoutesViewModel.kt:111]: Is there a test for this line?

**Review by cafesilencio — APPROVED** [review]

**Comment by abhishek1508** [comment]
I have been waiting the entire day for this [CI](https://app.circleci.com/pipelines/github/mapbox/mapbox-navigation-android/16992/workflows/b0d2dae0-0037-405e-984c-823b32406b04) to finish and ultimately had to kill it after 5h to merge it so we could move forward.

/cc @LukasPaczos @Guardiola31337

**Pull request merged**